docs/DeveloperGuide.md

```diff
@@ -8,23 +8,128 @@
+to print the ```Income``` added.
+
+![AddIncomeCommand](images/AddIncomeCommand_Sequence_Diagram.png)
+### ListExpenseCommand
+The listExpenseCommand is facilitated by ```Parser```, ```ListExpenseCommand``` and ```ExpenseList```.
+
+1. The user inputs the command to list expense. This input is handled by```Parser``` which returns the
+   ```listExpenseCommand``` if successful.
+2. ```ListExpenseCommand``` will then call its ```execute``` method
+   which makes use of ```System``` to print a new line ```"Expenses:"```.
+3. The ```ExpenseList``` method ```printExpenseList```
+   is called, which iterates through the expenseList, ```expenses``` and prints the index as well as a completed string of
+   expenses in ```expenses```.
+
+![ListExpenseCommand](images/ListExpenseCommand_Sequence_Diagram.png)
+### SetTargetCommand
+The setTargetCommand is facilitated by ```ChChing```, ```Parser```, ```TargetParser```, ```TargetStorage```.
+When the command receives to set target, the ```parse()``` method of the ```Parser``` object will read in the command and call the ```parseTarget()``` method from the TargetParser class 
+to instantiate a Target object and initializes a target value. Afterwards, ```parse()``` method calls the ```AddTargetCommand``` and  instantiate a ```SetTargetCommand``` object and returns to the ```Parser``` object.
+The ```Parser``` object then returns to ```ChChing```. ```ChChing``` object then runs the ```execute()``` method of the SetTargetCommand object. The method then calls the ```setTarget()``` method of the ```TargetStorage``` object to store the previously initialized ```Target``` object.
+
+![SetTargetCommand UML](images/SetTargetCommand.png)
+### SetCurrencyCommand
+The setCurrencyCommand is facilitated by ```System```, ```Selector```,  ```UI``` and ```ExpenseList```.
+The command receives the instruction from ```UI``` and will call the ```execute``` method.
+The ```execute()``` method in setCurrencyCommand will then call the ```containsCurrency(currency)``` method from ```Selector```.
+If the method returns false, which indicates that the currency is not available, the command will throw a new ChChingException and print ```"Currency not available"```.
+If the method returns true, the command will continue to set the currency in the selector hashmap to true.
+Afterwards, the ```execute()``` method will call the ```printSelector()``` method from ```Selector```.
+The ```printSelector()``` method will print all the available currencies in the selector hashmap.
+The selected currencies will be marked with a ```[X]``` and the unselected currencies will be marked with a ```[ ]```.
+
+![SetCurrencyCommand_sequence_diagram.png](images%2FSetCurrencyCommand_sequence_diagram.png)
 
 ## Product scope
+
 ### Target user profile
 
-{Describe the target user profile}
+Target users are people who are keen on improving their financial accountability
```

Comment:
Could this be elaborated on further? For example, what exactly is financial accountability?